src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -215,7 +215,12 @@ public IRecipeStorage getFirstRecipe(final Predicate<ItemStack> stackPredicate)
                 final ItemStorage checkItem = foo.getKey().getCleanedInput().stream().max(Comparator.comparingInt(ItemStorage::getAmount)).get();
                 candidates.put(foo.getKey(), getWarehouseCount(checkItem));
             }
-            return candidates.entrySet().stream().min(Map.Entry.comparingByValue(Comparator.reverseOrder())).get().getKey();
+            firstFound = candidates.entrySet().stream().min(Map.Entry.comparingByValue(Comparator.reverseOrder())).get().getKey();
```

Comment:
Either we should "break;" at first found, or do something else, because we override firstfound with each iteration.

Reply:
In the scan loop, we check against null before setting it, This highlighted one is in an "If", after populating counts, so we're only overriding it once in case we're adjusting for warehouse inventory. 
This is followed by another if, where we check for a multi-output, and convert it to a simple if so. 

That's a total of 3 sets maximum.

Reply:
But then "firstFound" is not exactly the right name, right?

Reply:
Fair point